addon/mixins/route.js

```diff
@@ -283,17 +281,78 @@ export default Ember.Mixin.create({
 
    @method updateInfinityModel
    @param {Ember.Enumerable} newObjects The new objects to add to the model
-   @return {Ember.Array} returns the updated infinity model
+   @return {Ember.Array} returns the new objects
    */
   updateInfinityModel(newObjects) {
-    var infinityModel = this.get(this.get('_modelPath'));
+    if (this.get('_firstPageLoaded')) {
+      let infinityModel = this._infinityModel();
+      return infinityModel.pushObjects(newObjects.get('content'));
+    }
 
-    return infinityModel.pushObjects(newObjects.get('content'));
+    return newObjects;
```

Comment:
on this first page, this array becomes the infinityModel that is returned on subsequent calls